docs/src/main/tut/docs/alternative_effects.md

```diff
@@ -11,62 +11,105 @@ title: "Side effects"
 
 ```scala
 import neotypes.implicits.mappers.results._ // Brings the implicit ResultMapper[String] instance into the scope.
-import neotypes.implicits.syntax.session._ // Provides the asScala[F[_]] extension method.
 import neotypes.implicits.syntax.string._ // Provides the query[T] extension method.
 import scala.concurrent.Future
+import scala.concurrent.ExecutionContext.Implicits.global
 
-val s = driver.session().asScala[Future]
+val program: Future[String] = for {
+  driver <- GraphDatabase.driver[Future]("bolt://localhost:7687", AuthTokens.basic("neo4j", "****"))
+  session <- driver.session()
+  data <- "match (p:Person {name: 'Charlize Theron'}) return p.name".query[String].single(session)
+  _ <- session.close()
+  _ <- driver.close()
+} yield data
 
-Await.result("match (p:Person {name: 'Charlize Theron'}) return p.name".query[String].single(s), 1 second)
+val data: String = Await.result(program, 1 second)
 ```
 
-### cats.effect.Async[F]
+### cats.effect.neotypes.Async[F]
 
 ```scala
-import cats.effect.IO
-import neotypes.cats.effect.implicits._ // Brings the implicit Async[IO] instance into the scope.
+import cats.effect.{IO, Resource}
+import neotypes.Session
+import neotypes.cats.effect.implicits._ // Brings the implicit neotypes.Async[IO] instance into the scope.
 import neotypes.implicits.mappers.results._ // Brings the implicit ResultMapper[String] instance into the scope.
-import neotypes.implicits.syntax.session._ // Provides the asScala[F[_]] extension method.
 import neotypes.implicits.syntax.string._ // Provides the query[T] extension method.
 
-val s = driver.session().asScala[IO]
+val session: Resource[IO, Session[IO]] = for {
+  driver <- Resource.make(
+    GraphDatabase.driver[IO]("bolt://localhost:7687", AuthTokens.basic("neo4j", "****"))
+  ) { driver => driver.close() }
```

Comment:
I'd probably simplify it as it's always kind of the same `(_.close)`

Reply:
And maybe `flatMap` is more idiomatic here as you only require one (plus there's a variable shadowing):

```scala
val session: Resource[IO, Session[IO]] = 
  Resource.make(
    GraphDatabase.driver[IO]("bolt://localhost:7687", AuthTokens.basic("neo4j", "****"))
  )(_.close).flatMap { driver =>
    Resource.make(driver.session)(_.close)
  }
```

Reply:
Since these are mainly style issues, I do not think we could get into a agreement. And since it is documentation, I wouldn't give it too much importance, since everyone will have his/her own flavor.

For example, IMHO, the `()` are important, since all this actions are side-effects _(even if they are now pure values, it is my way to immediately distinguish from pure functions and IO wrapped side effects)_.
Also, I have come to really avoid the `_.method` syntax, no matter how _"obvious"_ it is on what I am calling the method, when i have to come back to the code, it always takes me some time to figure it out _(maybe my only exception are things like  `c.foldLeft(M.empty)(_ |+| _)`)_.

But again, those are just my preferences, not need to agree on these.
However, I do think the examples are a little bit easier to read as they are.
_(but since I wrote them, I really can not vote for that 😅)_

@dimafeng What do you think?

Reply:
I agree it's mainly a style diff. 

I disagree in believing `()` is a *side effect*. It shouldn't be in these cases and that's why I prefer to design the interface without it as in

```scala
trait Driver[F[_]] {
  def close: F[Unit] // No () here
}
```

`()` represents a *side effect* when the API is impure IMO.

Reply:
Uhm... I have been checking other libraries like `fs2` and you are right, it should be without the parenthesis. I have to admit it looks awkward to me, but prefer to follow conventions.
Will make this changes as well as the others after @dimafeng leaves his feedback to make them all at once.

Reply:
I don't have strong opinion on `()`. I'd follow practices from **fs2** & **http4s**